_posts/2016-06-19-graphql-for-mobile.markdown

```diff
@@ -0,0 +1,99 @@
+---
+layout: post
+title: "GraphQL for iOS Developers"
+date: 2016-06-19 12:09
+author: orta
+categories: [mobile, graphql, eigen]
+---
+
+GraphQL is something you may have heard in passing, usually from the web team. It's a Facebook API technology, that describes itself as a _A Data Query Language and Runtime_. As mobile engineers, we can consider it an API, where the front-end team have as much control as the backend.
+
+This blog post covers our usage of GraphQL, and what I've learned in the last 3 months of using it in [Eigen](https://github.com/artsy/eigen/).
+
+<!-- more -->
+
+### So what is GraphQL
+
+So, you can get the full explaination on [their website](http://graphql.org). Though I found running through [Learn GraphQL](https://learngraphql.com) to really hammer down how it works as an API consumer. Reading the [introduction blog post](https://facebook.github.io/react/blog/2015/05/01/graphql-introduction.html) can be useful too, I'll be doing a quick TLDR though.
+
+GraphQL is an API middle-layer. It acts as an intermediate layer between multiple front-end clents and multiple back-end APIs. This means it can easily coalesce multiple API calls into a single request, this can be a _massive_ user experience improvement when you have a screen that requires information from multiple sources before you can present anything to a user.
```

Comment:
I think he means user-experience, which would be about saving on:

* the amount of data that needs to be downloaded (just what you need), because mobile networks are slow
* the number of network requests that need to be made, because mobile networks have unreliably high latency. This is especially true when fetching from different hosts, as you can’t even re-use connections in those cases.

I agree that this could use more explicit explanation. Some is given down below, but it’s also slightly hand-wavey https://github.com/artsy/artsy.github.io/pull/231#discussion-diff-68173704R40